1 table.sql

```diff
@@ -0,0 +1,59 @@
+    DepartureTime datetime not null,
+    ArrivalAirport char(3) not null
+        constraint ck_Booking_arrival_airport_must_be_3_letters check(ArrivalAirport like '[a-z][a-z][a-z]'),
+    ArrivalCountry varchar(25) not null
+        constraint ck_Booking_arrival_country_cannot_be_blank check(ArrivalCountry <> ''),
+    ArrivalTime datetime not null,
+    PassengerName varchar(50) not null
+        constraint ck_Booking_passenger_name_cannot_be_blank check(PassengerName <> ''),
+    PassengerDOB date not null,
+    PassengerAddress varchar(100) not null
+        constraint ck_Booking_passenger_address_cannot_be_blank check(PassengerAddress <> ''),
+    BookedDate date null,
+    PassportNum varchar(9) null
+        constraint ck_Booking_passport_num_cannot_start_with_0 check(PassportNum not like '0%'),
+        constraint ck_Booking_passport_num_length_must_be_9 check(len(PassportNum) = 9),
+        constraint ck_Booking_passport_num_can_only_be_numbers check(PassportNum not like '%[^0-9]%'),
+    PassportIssueDate date null,
+    PassportExpiryDate as case 
+        when PassportIssueDate >= dateadd(year, 16, PassengerDOB) then dateadd(year, 9, dateadd(month, 6, PassportIssueDate))
+        else dateadd(year, 5, PassportIssueDate)
+    end persisted,
+    PassportNationality varchar(25) null
+        constraint Booking_passport_nationality_cannot_be_blank check(PassportNationality <> ''),
+    CheckedInTime datetime null,
+    
+    -- Date constraints
+	constraint ck_Booking_passport_issue_date_cannot_be_before_passenger_dob check(PassengerDOB <= PassportIssueDate),
+	constraint ck_Booking_passport_booked_date_cannot_be_before_passport_issue_date check(PassportIssueDate <= BookedDate),
+	constraint ck_Booking_checked_in_time_cannot_be_before_booked_date check((BookedDate <= CheckedInTime) or CheckedInTime is null),
+    constraint ck_Booking_passenger_age_must_be_between_16_and_90 
+		check(DepartureTime between dateadd(year, 16, PassengerDOB) and dateadd(year, 91, PassengerDOB)),
+    constraint ck_Booking_booked_date_must_be_between_1_year_and_1_hour_before_departure_time 
+		check(BookedDate between dateadd(year, -1, DepartureTime) and dateadd(hour, -1, DepartureTime)),
+    constraint ck_Booking_checked_in_time_must_be_between_30_days_and_1_hour_before_departure_time 
+		check(CheckedInTime between dateadd(day, -30, DepartureTime) and dateadd(hour, -1, DepartureTime)),
+    constraint ck_Booking_arrival_time_must_be_after_departure_time check(ArrivalTime > DepartureTime),
+	constraint ck_passport_is_or_will_be_expired_on_departure_date
+		check(PassportExpiryDate > DepartureTime or ArrivalCountry in (DepartureCountry, PassportNationality)),
+    
+	--All or non constraints
+	constraint ck_Booking_passport_num_passport_issue_date_passport_nationality_and_checked_in_time_must_all_either_be_completed_or_null
+        check((PassportNum is null and PassportIssueDate is null and PassportNationality is null and CheckedInTime is null)
```

Comment:
You can have passport details even if not checked in yet.

Reply:
"Aren't they working hand in hand? According to the [spec](https://github.com/CPU-Code-School/cpu-biz-scenario/blob/05275ffa43384c53348a84e25fbb3cf1878907f9/airline.sql#L6C157-L6C157), 'When you book the flight you would only need to provide your name, DOB, and address, but in order to be checked in and travel, we need the passport details.' So, I understood that when you check in, you give your passport details, and it's one process."

Reply:
Not necessarily. You can add the passport details before. At the time you check in, you **must** have the details.